src/Inciter/AMR/refinement.h

```diff
@@ -0,0 +1,779 @@
+                tet_store->edge_store.generate(BD, BC, Edge_Lock_Case::unlocked);
+                tet_store->edge_store.generate(CD, BC, Edge_Lock_Case::unlocked);
+
+                // Outside edges for face ABD
+                tet_store->edge_store.generate(AD, BD, Edge_Lock_Case::unlocked);
+                tet_store->edge_store.generate(AB, AD, Edge_Lock_Case::unlocked);
+                tet_store->edge_store.generate(AB, BD, Edge_Lock_Case::unlocked);
+
+                // Interior Edges
+                tet_store->edge_store.generate(AC, BD, Edge_Lock_Case::unlocked);
+                tet_store->edge_store.generate(CD, AD, Edge_Lock_Case::unlocked);
+
+                // Add the new tets
+                //
+                // External
+                // A AB AC AD - A
+                // B BA BC BD - B
+                // C CA CB CD - C
+                // D DA DB DC - D
+                // -
+                // Internal (for a face BDC, it's the intermediat and mid opposite)
+                // BC CD DB AC - BDC
+                // AB BD AD AC - ABD
+                // AB AC BC BD - ABC
+                // AC AD CD BD - ACD
+                //
+
+                // TODO: This is actually generating IDs not trying to get them
+                child_id_list_t child = generate_child_ids(tet_id);
+
+                // This order should give a positive Jacobian
+                tet_store->add(child[0], A, AB, AC, AD, Refinement_Case::one_to_eight, tet_id);
+                tet_store->add(child[1], B, BC, AB, BD, Refinement_Case::one_to_eight, tet_id);
+                tet_store->add(child[2], C, AC, BC, CD, Refinement_Case::one_to_eight, tet_id);
+                tet_store->add(child[3], D, AD, CD, BD, Refinement_Case::one_to_eight, tet_id);
+
+                tet_store->add(child[4], BC, CD, AC, BD, Refinement_Case::one_to_eight, tet_id);
+                tet_store->add(child[5], AB, BD, AC, AD, Refinement_Case::one_to_eight, tet_id);
+                tet_store->add(child[6], AB, BC, AC, BD, Refinement_Case::one_to_eight, tet_id);
+                tet_store->add(child[7], AC, BD, CD, AD, Refinement_Case::one_to_eight, tet_id);
+
+                tet_store->deactivate(tet_id);
+
+            }
+
+            // This is just a simple assignment, but I wanted to abstract it
+            // for if we change the underlying type to something which a simple
+            // assignment is no longer safe
+            /**
+             * @brief Function to duplicate (deep copy) a tet. Useful for when
+             * you want to make a tet that's very similar to an existing one
+             *
+             * @param out The tet to store the copy
+             * @param original The tet to copy the data from
+             */
+            void copy_tet(tet_t* out, tet_t* original)
+            {
+                // NOTE: This will do a deep copy, so is safer than it may look
+                *out = *original;
+            }
```

Comment:
I think the best is, well, to not do anything. ;-) And let the compiler auto-generate the copy constructor, copy assignment, as well as the move constructor and the move assignment for `std::array<>`.